.github/workflows/notify-merge.js

```diff
@@ -0,0 +1,123 @@
+const fs = require('fs');
+const { slackNotification } = require('./helpers.js');
+
+const SLACK = {
+    merge: ({ html_url, number, title, prefix = '' }) =>
+        `:merged: PR merged to stage: ${prefix} <${html_url}|#${number}: ${title}>.`,
+};
+
+const mergeRegex = /Merge pull request #(\d+)/i;
+function getRepoInfo() {
+    const repoFull = process.env.GITHUB_REPOSITORY;
+    if (!repoFull) {
+        throw new Error("GITHUB_REPOSITORY is not defined in the environment.");
+    }
+    const [owner, repo] = repoFull.split('/');
+    return { owner, repo };
+}
+
+async function getPR(prNumber, owner, repo, token) {
+    const url = `https://api.github.com/repos/${owner}/${repo}/pulls/${prNumber}`;
+    const response = await fetch(url, {
+        headers: {
+            Accept: 'application/vnd.github.v3+json',
+            Authorization: `token ${token}`,
+        },
+    });
+    if (!response.ok) {
+        throw new Error(`Failed to fetch PR ${prNumber}: ${response.status}`);
+    }
+    return await response.json();
```

Comment:
You already await the Promise from the calling function
```suggestion
    return response.json();
```